src/JSInterop/Microsoft.JSInterop/test/Infrastructure/DotNetDispatcherTest.cs

```diff
@@ -314,6 +314,110 @@ public async Task EndInvoke_WithNullError()
         Assert.Empty(ex.Message);
     }
 
+    [Fact]
+    public void EndInvokeJS_DoesNotThrowJSONExceptionIfTaskCancelled()
+    {
+        // Arrange
+        var jsRuntime = new TestJSRuntime();
+        var testDTO = new TestDTO { StringVal = "Hello", IntVal = 4 };
+        var cts = new CancellationTokenSource();
+        var argsJson = JsonSerializer.Serialize(new object[] { jsRuntime.LastInvocationAsyncHandle, true, testDTO }, jsRuntime.JsonSerializerOptions);
+
+        // Act
+        var task = jsRuntime.InvokeAsync<TestDTO>("unimportant", cts.Token);
+
+        cts.Cancel();
+
+        DotNetDispatcher.EndInvokeJS(jsRuntime, argsJson);
+
+        // Assert
+        Assert.False(task.IsCompletedSuccessfully);
+        Assert.True(task.IsCanceled);
+    }
+
+    [Fact]
+    public void EndInvokeJS_ThrowsIfJsonIsEmptyString()
```

Comment:
Remaining tests have been moved up from later in the file (so all the `EndInvokeJS` tests remain in one section). The `Arrange` section was also added for the tests.